src/openassetio-core/specification/Specification.cpp

```diff
@@ -22,7 +22,7 @@ class Specification::Impl {
     TraitIds ids;
     ids.reserve(data_.size());
     for (const auto& item : data_) {
-      ids.push_back(item.first);
+      ids.insert(item.first);
```

Comment:
Consider using structured bindings here which I think is a little more readable:

      for (const auto& [key, value] : data_) {
        ids.insert(key);
      }

Reply:
@feltech and had discussed this, and went for this one, as otherwise you have an unused `value` binding that triggers other warnings...

Reply:
I see, makes sense.

Reply:
Do you think it's worth a comment?

Reply:
I didn't realize that it would lead to warnings, which is a valid point. I don't think it's worth littering the code with comments about this, or at least I can't come up with something that's short and to the point. 

Maybe this might work? I've not tried it in the sanitizer yet.

    for ([[maybe_unused]] const auto& [key, value] : data) {

Reply:
Cool, we can pick this up as another PR maybe, as its out of scope for the change to `unordered_map`?

Reply:
Yes, definitely not a deal breaker for this PR.